mesonbuild/mtest.py

```diff
@@ -1606,9 +1606,12 @@ def load_metadata(self) -> None:
             # happen before rebuild_deps(), because we need the correct list of
             # tests and their dependencies to compute
             if not self.options.no_rebuild:
-                ret = subprocess.run(self.ninja + ['build.ninja']).returncode
-                if ret != 0:
-                    raise TestException(f'Could not configure {self.options.wd!r}')
+                teststdo = subprocess.run(self.ninja + ['-n', 'build.ninja'], capture_output=True, text=True).stdout
+                if 'ninja: no work to do.' not in teststdo and 'samu: nothing to do' not in teststdo:
+                    stdo = sys.stderr if self.options.list else sys.stdout
+                    ret = subprocess.run(self.ninja + ['build.ninja'], stdout=stdo.fileno())
+                    if ret.returncode != 0:
+                        raise TestException(f'Could not configure {self.options.wd!r}')
```

Comment:
Nitpicking, and was already like that, but shouldn't this be "reconfigure" or "rebuild"?